src/helm/clients/cohere_client.py

```diff
@@ -27,92 +27,57 @@ def make_request(self, request: Request) -> RequestResult:
         ), f"Invalid top_k_per_token: {request.top_k_per_token}. Valid range: [0..500]"
-        # p: "Set to 1.0 or 0 to disable. If set to a probability 0.0 < p < 1.0,
-        #     it ensures that only the most likely tokens, with total probability mass of p."
-        assert 0.0 <= request.top_p <= 1.0, f"Invalid top_p: {request.top_p}. Valid range: [0,1]"
-        # frequency_penalty: "min value of 0.0, max value of 1.0"
-        assert (
-            0.0 <= request.frequency_penalty <= 1.0
-        ), f"Invalid frequency_penalty: {request.frequency_penalty}. Valid range: [0,1]"
-        # presence_penalty: "min value of 0.0, max value of 1.0"
-        assert (
-            0.0 <= request.presence_penalty <= 1.0
-        ), f"Invalid presence_penalty: {request.presence_penalty}. Valid range: [0,1]"
 
+        assert 0.0 <= request.top_p <= 1.0, f"Invalid top_p: {request.top_p}. Valid range: [0,1]"
+    
         raw_request = {
-            "prompt": request.prompt,  # Note that "trailing whitespaces [of prompts] will be trimmed".
-            "max_tokens": request.max_tokens,
             "model": request.model_engine,
+            "message": request.prompt,
+            "max_tokens": request.max_tokens,
+            "prompt_truncation": "OFF",
             "temperature": request.temperature,
-            "num_generations": request.num_completions,
             "k": request.top_k_per_token,
             "p": request.top_p,
             "frequency_penalty": request.frequency_penalty,
             "presence_penalty": request.presence_penalty,
             "stop_sequences": request.stop_sequences,
-            "return_likelihoods": return_likelihoods,
         }
 
         try:
 
             def do_it():
                 """
                 Send the request to the Cohere Generate API. Responses will be structured like this:
-                {
-                    "generations": [
-                        {
-                            "text": string,
-                            "likelihood": float,
-                            "token_likelihoods": [{"token": string, "likelihood": float}, ...]
-                        },
-                        ...
-                    ]
+                cohere.Chat {
+                    message: What's up?
+                    text: Hey there! How's it going? I'm doing well, thank you for asking 😊.
+                    ...
                 }
-
-                Note: The stop reason is not included in the response.
                 """
                 # Cohere has a Python SDK, but it requires additional post-processing to convert their response
                 # objects (`Generations`) to JSON, the form the cache expects the responses to be in.
                 response = requests.request(
                     method="POST",
-                    url=get_cohere_url(CohereClient.GENERATE_ENDPOINT),
+                    url=get_cohere_url(CohereClient.CHAT_ENDPOINT),
```

Comment:
Good point! I do think some of the older models are no longer online

I think going forward we will only be supporting `chat` and not `generate` for most models

Reply:
I am curious if we could even remove some old Cohere stuff.

In the latest version of this PR, I explored some deletions we can make, such as tokenizer removals. Let me know what you think!

Reply:
Regarding checking what breaks - you should be able to run e.g. `cohere/command-light` and that should still work.

I'm inclined to keep the old text API endpoint support, so that people can still use them for reproducing old results (at least until the text API endpoint is gone).

As for the tokenizer, can the command-r tokenizer be used for the older command models, or do those use a different tokenizer? Also, is there still a (small) length limit on the tokenizer API? If either of those assumptions are no longer true, it would simplify a lot of HELM code here.